lib/hiera/backend/module_data_backend.rb

```diff
@@ -68,13 +60,13 @@ def lookup(key, scope, order_override, resolution_type)
 
         unless module_name
           Hiera.debug("Skipping Module Data backend as this does not look like a module")
-          return no_answer
+          return answer
```

Comment:
bit worried about this and the one on line 69 too.  It should throw :no_such_key in all cases that doesnt result in a answer, so here it will return nil and that will mess things up.

So I would leave the old `no_answer` method as is, and here and 69 do `return no_answer` just like that.  and on 102 do `return no_answer unless found`

this will ensure that the right thing is thrown in all these cases.